bioconda.py

```diff
@@ -1,49 +1,86 @@
 """Bioconda-related functionality"""
 
-from utils import clean_github_link, find_all_paths
+import os
+import logging
 
+from bioconda_utils.recipe import Recipe
+from utils import (
+    find_all_paths,
+    clean_github_link,
+    nested_dictionary_extract,
+    flatten_list_with_lists,
+)
 
-def generate_bioconda_source_links(filepath):
+
+logger = logging.getLogger(__name__)
+
+
+def generate_bioconda_source_links(dirpath):
     """Create list of of all bioconda-related links in directory.
 
     Can be used to create a list of all bioconda package source links
     in the bioconda registry.
 
     Args:
-       filepath (str): filepath to a dir with 1 or more package.tomls
+       dirpath (str): dirpath to a bioconda recipe directory with 1 or
+           more meta.yaml files
 
     Returns:
         list - all source links from all discovered meta.yaml files
     """
-    links = []
-    paths = find_all_paths(path_endings=["meta.yaml", "meta.yml"], base=filepath)
+    # Using set() to ensure no duplicates
+    all_links = set()
+    logger.info(f"Generating Bioconda source links from {dirpath}")
+    paths = find_all_paths(path_endings=["meta.yaml", "meta.yml"], base=dirpath)
     for path in paths:
-        link = parse_meta_dot_yaml_for_source_link(path)
-        links.append(link)
+        links = parse_meta_dot_yaml_for_source_links(path)
+        for link in links:
+            all_links.add(link)
 
-    return links
+    return list(all_links)
 
 
-def parse_meta_dot_yaml_for_source_link(filepath):
-    """Find source code link in meta.yaml file.
+def parse_meta_dot_yaml_for_source_links(filepath):
+    """Find source code GitHub links in meta.yaml file.
```

Comment:
Technically the links are not necessarily to GitHub. They could be to GitLab, for instance. Hmmm. Perhaps we need a little re-engineering. Perhaps we discuss tomorrow.